File: DoAnCoSo Ver 14.0/DoAnCoSo/Models/ServiceRecordBlock.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DoAnCoSo.Models
{
    public class ServiceRecordBlock
    {
        [Key]
        public int BlockId { get; set; }

        public int RecordId { get; set; }
        [ForeignKey("RecordId")]
        public PetServiceRecord PetServiceRecord { get; set; }

        [Required]
        public string CurrentHash { get; set; }

        public string? PreviousHash { get; set; }

        public DateTime Timestamp { get; set; }
    }
}

File: AppSpaHomestayChoThuCung/lib/model/service/service_record_block.dart
class ServiceRecordBlockModel {
  final int blockId;
  final int recordId;
  final String currentHash;
  final String? previousHash;
  final DateTime timestamp;

  ServiceRecordBlockModel({
    required this.blockId,
    required this.recordId,
    required this.currentHash,
    this.previousHash,
    required this.timestamp,
  });

  factory ServiceRecordBlockModel.fromJson(Map<String, dynamic> json) {
    return ServiceRecordBlockModel(
      blockId: json['blockId'],
      recordId: json['recordId'],
      currentHash: json['currentHash'],
      previousHash: json['previousHash'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }
}